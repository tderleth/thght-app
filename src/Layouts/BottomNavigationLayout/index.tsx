import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import BottomNavigation from '../../components/BottomNavigation';

export interface LayoutProps {
  children?: React.ReactNode;
}

const styles = StyleSheet.create({
  layout: {
    flex: 1,
    flexDirection: 'column',
  },
  bottomNav: {
    flex: 1,
    backgroundColor: 'white',
  },
  content: {
    backgroundColor: 'black',
    flex: 9,
    justifyContent: 'center',
  },
});

const BottomNavigationLayout: React.SFC = ({ children }: LayoutProps): JSX.Element => (
  <View style={styles.layout}>
    <View style={styles.content}>{children}</View>
    <View style={styles.bottomNav}>
      <BottomNavigation />
    </View>
  </View>
);

export default BottomNavigationLayout;
