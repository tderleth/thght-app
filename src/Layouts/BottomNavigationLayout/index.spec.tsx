import * as React from 'react';
import { MemoryRouter } from 'react-router-native';
import * as renderer from 'react-test-renderer';
import BottomNavigationLayout from '.';

describe('CaptureThought', () => {
  it('snapshot test', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <BottomNavigationLayout />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
