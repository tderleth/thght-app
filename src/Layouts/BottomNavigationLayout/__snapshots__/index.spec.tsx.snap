// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptureThought snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "flex": 9,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
              }
            }
          >
            CaptureThought
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
              }
            }
          >
            Reflect
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
